language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs

```diff
@@ -0,0 +1,221 @@
+-- Copyright (c) 2020 The DAML Authors. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module DA.Test.Daml2Ts (main) where
+
+import Control.Monad.Extra
+import System.FilePath
+import System.IO.Extra
+import System.Environment.Blank
+import System.Directory.Extra
+import System.Process
+import System.Exit
+import DA.Bazel.Runfiles
+import Test.Tasty
+import Test.Tasty.HUnit
+
+import SdkVersion
+
+main :: IO ()
+main = do
+    setEnv "TASTY_NUM_THREADS" "1" True
+    damlc <- locateRunfiles (mainWorkspace </> "compiler" </> "damlc" </> exe "damlc")
+    daml2ts <- locateRunfiles (mainWorkspace </> "language-support" </> "ts" </> "codegen" </> exe "daml2ts")
+    davl <- locateRunfiles ("davl" </> "released")
+    defaultMain $ tests damlc daml2ts davl
+
+tests :: FilePath -> FilePath -> FilePath -> TestTree
+tests damlc daml2ts davl = testGroup "daml2Ts"
+  [ testCaseSteps "Breathing test" $ \step -> withTempDir $ \tmpDir -> do
+      let grover = tmpDir </> "grover"
+      let groverTs = tmpDir </> "grover-ts"
+      let groverDar = grover </> ".daml" </> "dist" </> "grover-1.0.dar"
+      step "Creating project 'grover'..."
+      createDirectoryIfMissing True (grover </> "daml")
+      createDirectoryIfMissing True groverTs
+      writeFileUTF8 (grover </> "daml" </> "Grover.daml") $ unlines
+        [ "daml 1.2"
+        , "module Grover where"
+        , "template Grover"
+        , "  with puppeteer : Party"
+        , "  where"
+        , "    signatory puppeteer"
+        , "    choice Grover_GoSuper: ContractId Grover"
+        , "      controller puppeteer"
+        , "      do"
+        , "        return self"
+        ]
+      writeFileUTF8 (grover </> "daml.yaml") $ unlines
```

Comment:
I wonder if we should finally invest in some shared helpers to create all these `daml.yaml` files and the directories containing them. The packaging tests are already full of these repetitions...

Reply:
Yes, there's a mini-library emerging. Maybe not today but soon.